### PR TITLE
Enforce -D warnings on tests

### PR DESCRIPTION
**Pull request by CAD97**
We had a lot of `Pair::into_span` hanging around, including some `.clone().into_span()`.

Additionally, the code compiled for `cargo bootstrap` isn't clippy-checked, so that will need addressing at a later point.

r? @dragostis

### CODE REVIEW

File: pest/src/iterators/pair.rs [hunk 1]
```diff
@@ -135,7 +135,7 @@ impl<'i, R: RuleType> Pair<'i, R> {
     /// #     state.rule(Rule::ab, |s| s.match_string("ab"))
     /// }).unwrap().next().unwrap();
     ///
-    /// assert_eq!(pair.into_span().as_str(), "ab");
+    /// assert_eq!(pair.as_span().as_str(), "ab");
```
Comment (CAD97) [inline pest/src/iterators/pair.rs]: This got caught up in my refactoring (the doc-test for `Pair::into_span`). I'm unsure which we should use here.

Comment (dragostis) [reply pest/src/iterators/pair.rs]: I'd leave the old one here.

**Comment by dragostis** [comment]
@CAD97, would you still be interested to fix the remaining issues with the PR?

**Comment by MarinPostma** [comment]
closing since this is inactive and should be fixed with #507

**Comment by CAD97** [comment]
Oh wow this was still hanging around 😅

**Pull request closed without merging**